browser/ui/brave_browser.cc

```diff
@@ -128,6 +131,33 @@ void BraveBrowser::TabStripEmpty() {
                                 true, std::nullopt));
 }
 
+void BraveBrowser::RunFileChooser(
+    content::RenderFrameHost* render_frame_host,
+    scoped_refptr<content::FileSelectListener> listener,
+    const blink::mojom::FileChooserParams& params) {
+#if BUILDFLAG(IS_ANDROID)
+  Browser::RunFileChooser(render_frame_host, listener, params);
+#else
+  auto new_params = params.Clone();
+  if (new_params->title.empty()) {
+    // Fill title of file chooser with origin of the frame.
+
+    // Note that save mode param is for PPAPI. 'Save As...' or downloading
+    // something doesn't reach here. They show 'select file dialog' from
+    // DownloadFilePicker::DownloadFilePicker directly.
+    // https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/public/mojom/choosers/file_chooser.mojom;l=27;drc=047c7dc4ee1ce908d7fea38ca063fa2f80f92c77
+
+    new_params->title = brave::GetFileSelectTitle(
+        content::WebContents::FromRenderFrameHost(render_frame_host),
+        render_frame_host->GetLastCommittedOrigin(),
+        params.mode == blink::mojom::FileChooserParams::Mode::kSave
```

Comment:
I think this need to be refactored into `bool is_save` or `enum class FileSelectTitleType { kOpen, kSave };`. This way you don't need to export ID maps, you can even store them as constexpr maps internally in `.cc`.

wdyt?

Reply:
I think having `enum class FileSelectTittleType` could be a good idea, but there's one more reason we have resource id maps parameter. As you suggested before, we need to compare `brave::GetFileSelectTitle` and `AppModalDialogManager::GetSiteFrameTitle`. So we should be able to specify resource ids to use.

Reply:
how about adding something like `FileSelectTittleType::kChromiumDefault` for tests?

Reply:
That's a good idea! Done.